### PR TITLE
Added spinner option to input component.

### PR DESCRIPTION
**Pull request by felipeArnold**
Good afternoon, how are you?
I found a need to add a spinner to the input component, in my case I needed to consult an api, with the spinner I can show the user that their request is being consulted.

image

### CODE REVIEW

File: resources/views/components/input.blade.php [hunk 1]
```diff
@@ -52,19 +51,35 @@ class="w-5 h-5"
...
+                    fill="none"
+                    viewBox="0 0 24 24"
+                    @if (preg_replace('/[^a-zA-Z]+/', '', $spinner))
+                        wire:target="{{ $spinner }}"
+                    @endif
+                    wire:loading.block
+                >
+                    <circle class="opacity-25" cx="12" cy="12" r="10" stroke="currentColor" stroke-width="4"></circle>
+                    <path class="opacity-75" fill="currentColor" d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"></path>
+                </svg>
```
Comment (ph7jack) [inline resources/views/components/input.blade.php]: Add one tab indentation here.

File: src/View/Components/Input.php [hunk 2]
```diff
@@ -42,20 +45,22 @@ public function __construct(
...
-        $this->append     = $append;
-        $this->errorless  = $errorless;
+        $this->icon = $icon;
+        $this->rightIcon = $rightIcon;
+        $this->prefix = $prefix;
+        $this->suffix = $suffix;
+        $this->prepend = $prepend;
+        $this->append = $append;
+        $this->errorless = $errorless;
+        $this->spinner = $spinner;
```
Comment (ph7jack) [inline src/View/Components/Input.php]: Please, use the phpcs fixer to format this file. Wireui is using it for code style.

**Review by ph7jack — CHANGES_REQUESTED** [review]
Thanks for contributing, your PR looks good!

File: resources/views/components/input.blade.php [hunk 3]
```diff
@@ -52,19 +51,35 @@ class="w-5 h-5"
...
+        @if ($suffix || $rightIcon || ($hasError && !$append) || $spinner)
             <div class="absolute inset-y-0 right-0 pr-2.5 flex items-center pointer-events-none
                 {{ $hasError ? 'text-negative-500' : 'text-secondary-400' }}">
+
                 @if ($rightIcon)
                     <x-dynamic-component
                         :component="WireUi::component('icon')"
                         :name="$rightIcon"
                         class="w-5 h-5"
+                        @if ($spinner) :wire:loading.remove="(bool) $spinner" @endif
```
Comment (ph7jack) [inline resources/views/components/input.blade.php]: Using directives inside a blade component attributes wont work

**Review by ph7jack — CHANGES_REQUESTED** [review]

**Review by ph7jack — APPROVED** [review]
thanks for contributing

**Comment by dwainscheeren** [comment]
I now get error `Undefined variable $spinner` on using 
```
<x-input 
                class="w-full" 
                placeholder="Zoeken..." 
                wire:model.live="term"
                type="search" 
                icon="search"
/>

**Pull request merged**